libmachine/provision/ubuntu_systemd.go

```diff
@@ -102,6 +102,19 @@ func (provisioner *UbuntuSystemdProvisioner) Provision(swarmOptions swarm.Option
 	provisioner.EngineOptions = engineOptions
 	swarmOptions.Env = engineOptions.Env
 
+	log.Debug("waiting for cloud-init to complete")
+	err := mcnutils.WaitFor(func() bool {
+		_, err := provisioner.SSHCommand("sudo cloud-init status --wait")
+		if err != nil {
+			log.Warnf("Failed to wait for cloud-init to complete: %v", err)
+			return false
+		}
+		return true
+	})
+	if err != nil {
+		return err
```

Comment:
Suggesting to wrap this error as it will return up something like https://github.com/docker/machine/blob/bd45ab13d88c32a3dd701485983354514abc41fa/libmachine/mcnutils/utils.go#L92 which doesn't make it clear what command we failed to wait for succeeding.